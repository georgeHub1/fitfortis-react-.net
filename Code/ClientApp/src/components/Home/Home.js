import React, { Component, lazy, Suspense } from 'react';
import { FormattedMessage, injectIntl } from 'react-intl';
import styles from './styles.module.less';
// import HomeNews from './HomeNews';
// import HomeCharts from './HomeCharts';
import { connect } from 'react-redux';
import SignUpPrompt from '../Common/SignUpPrompt';
import { localeSet } from '../../redux/locale.action';

import analyticId from '../../utils/AnalyticsAndAbTests.tsx';
const HomeNews = lazy(() => import('./HomeNews'));
const HomeCharts = lazy(() => import('./HomeCharts'));
const renderLoader = () => <p></p>;

class Home extends Component {
  displayName = Home.name

  componentDidMount () {
    window.scrollTop = 0;
    if (window.location.search.includes('?lang=')) {
      const lan = window.location.search.includes('?lang=') ? window.location.search.substring(6, 11) : (localStorage.getItem('language') || 'en');

      this.props.dispatch(localeSet(lan));
      localStorage.setItem('language', lan);
    }
    analyticId.firebaseAnalyticsLog('Home');
  }

  isDisable = () => {
    const { auth } = this.props;

    return !auth.loggenIn;
  }
  handleAddNewChart = () => localStorage.setItem('isShowAddNewChart', 'yes');

  render () {
    const { auth } = this.props;

    return (
      <div>
        <h1 className={`${styles.header} HideMobileTitle`}>
          <FormattedMessage id="Home.header" />
        </h1>
        {this.isDisable() ? <SignUpPrompt /> : ''}
        <div className={auth.loggenIn ? 'mainHomeSection' : 'mainHomeSection notLogin'}>
          <Suspense fallback={renderLoader()}>
            <HomeCharts />
          <div className="MainhomeLeft">
            <div className={styles.homeLeft}>
              <HomeNews history={this.props.history}/>
            </div>
          </div>
          </Suspense>
        </div>
      </div>
    );
  }
}
const mapStateToProps = ({ auth }) => ({ auth });

export default connect(mapStateToProps)(injectIntl(Home));
