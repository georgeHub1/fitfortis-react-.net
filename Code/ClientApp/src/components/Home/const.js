export const data = [
  {
    title: 'Headaches',
    description: 'The 21st century is the golden age for surgical procedures, among them an increasingly popular method of corrective eye surgery known as LASIK. LASIK is an acronym for the surgery, which stands for Laser Assisted In Situ Keratomileusis. This procedure has become increasingly reliable, with unresolved complications reported in, Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat. Duis aute irure dolor in reprehenderit in voluptate velit esse cillum dolore eu fugiat nulla pariatur.',
    date: new Date(),
    id: '0825586e-9bf4-43ba-53fb-08d77e21eb9f1'
  },
  {
    title: 'Injections',
    description: 'Among them an increasingly popular method of corrective eye surgery known as LASIK. LASIK is an acronym for the surgery, which stands for Laser Assisted In Situ Keratomileusis. This procedure has become increasingly reliable, with unresolved complications reported in, Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat. Duis aute irure dolor in reprehenderit in voluptate velit esse cillum dolore eu fugiat nulla pariatur.',
    date: new Date(),
    image: 'image',
    id: '0825586e-9bf4-43ba-53fb-08d77e21eb9f2'
  },
  {
    title: 'Headaches Injections',
    description: 'The 21st century is the golden age for surgical procedures, among them an increasingly popular method of corrective eye surgery known as LASIK. LASIK is an acronym for the surgery, which stands for Laser Assisted In Situ Keratomileusis. This procedure has become increasingly reliable, with unresolved complications reported in, Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat. Duis aute irure dolor in reprehenderit in voluptate velit esse cillum dolore eu fugiat nulla pariatur.',
    date: new Date(),
    image: 'image',
    id: '0825586e-9bf4-43ba-53fb-08d77e21eb9f3'
  },
  {
    title: 'Botox Injections',
    description: 'The 21st century is the golden age for surgical procedures, among them an increasingly popular method of corrective eye surgery known as LASIK. LASIK is an acronym for the surgery, which stands for Laser Assisted In Situ Keratomileusis. This procedure has become increasingly reliable, with unresolved complications reported in, Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat. Duis aute irure dolor in reprehenderit in voluptate velit esse cillum dolore eu fugiat nulla pariatur.',
    date: new Date(),
    image: 'image',
    id: '0825586e-9bf4-43ba-53fb-08d77e21eb9f4'
  },
  {
    title: 'And  Injections',
    description: 'The 21st century is the golden age for surgical procedures, among them an increasingly popular method of corrective eye surgery known as LASIK. LASIK is an acronym for the surgery, which stands for Laser Assisted In Situ Keratomileusis. This procedure has become increasingly reliable, with unresolved complications reported in, Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat. Duis aute irure dolor in reprehenderit in voluptate velit esse cillum dolore eu fugiat nulla pariatur.',
    date: new Date(),
    id: '0825586e-9bf4-43ba-53fb-08d77e21eb9f5'
  },
  {
    title: 'Botox',
    description: 'The 21st century is the golden age for surgical procedures, among them an increasingly popular method of corrective eye surgery known as LASIK. LASIK is an acronym for the surgery, which stands for Laser Assisted In Situ Keratomileusis. This procedure has become increasingly reliable, with unresolved complications reported in, Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat. Duis aute irure dolor in reprehenderit in voluptate velit esse cillum dolore eu fugiat nulla pariatur.',
    date: new Date(),
    image: 'image',
    id: '0825586e-9bf4-43ba-53fb-08d77e21eb9f6'
  },
  {
    title: 'Injections',
    description: 'The 21st century is the golden age for surgical procedures, among them an increasingly popular method of corrective eye surgery known as LASIK. LASIK is an acronym for the surgery, which stands for Laser Assisted In Situ Keratomileusis. This procedure has become increasingly reliable, with unresolved complications reported in, Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat. Duis aute irure dolor in reprehenderit in voluptate velit esse cillum dolore eu fugiat nulla pariatur.',
    date: new Date(),
    id: '0825586e-9bf4-43ba-53fb-08d77e21eb9f7'
  }
];

function compareDate (a, b) {
  const _a = a.date;
  const _b = b.date;

  let comparison = 0;

  if (new Date(_a) > new Date(_b)) {
  comparison = -1;
  } else if (new Date(_a) < new Date(_b)) {
  comparison = 1;
  }
  return comparison;
}

export const filter = (filterData, text) => {
  if (text) {
    return filterData.filter(x => {
      return x.title.toLowerCase().includes(text.toLowerCase());
    });
  }
  return filterData.sort(compareDate);
};
