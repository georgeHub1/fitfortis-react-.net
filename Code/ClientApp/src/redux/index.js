import { createStore, applyMiddleware } from 'redux';

import reducers from './index.reducer';
import middleware from './middleware.store';

import { defaultAuth } from './auth.reducer.js';
import { defaultCharts } from './metricsChart.reducer';
import { defaultMetricsDateRange } from './metricsDateRange.reducer.js';
import { defaultMetricsMetric } from './metricsMetric.reducer.js';
import { defaultMetricsMetricData } from './metricsMetricData.reducer.js';
import { defaultEncyclopedia } from './encyclopedia.reducer.js';
import { defaultProfile } from './profile.reducer.js';
import { defaultSymptomChecker } from './symptomChecker.reducer.js';
import { defaultDocument } from './document.reducer';
import { defaultItem } from './item.reducer.js';
import { defaultAdminTool } from './adminTool.reducer';
const createStoreWithMiddleware = applyMiddleware(...middleware)(createStore);

export const initialState = {
  auth: defaultAuth,
  metricsChart: defaultCharts,
  metricsDateRange: defaultMetricsDateRange,
  metricsMetric: defaultMetricsMetric,
  metricsMetricData: defaultMetricsMetricData,
  encyclopedia: defaultEncyclopedia,
  profile: defaultProfile,
  symptomChecker: defaultSymptomChecker,
  document: defaultDocument,
  item: defaultItem,
  adminTool: defaultAdminTool
};

const createAppStore = createStoreWithMiddleware(
  reducers,
  initialState,
  window.__REDUX_DEVTOOLS_EXTENSION__ && window.__REDUX_DEVTOOLS_EXTENSION__()
);

export default createAppStore;
