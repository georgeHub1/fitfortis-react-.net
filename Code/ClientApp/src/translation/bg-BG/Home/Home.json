{
  "header": "Начална страница",
  "addNewChart": "Добави графика"
}