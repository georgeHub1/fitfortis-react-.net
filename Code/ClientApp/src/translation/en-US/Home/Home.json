{
  "header": "Home",
  "addNewChart": "Add New Chart"
}