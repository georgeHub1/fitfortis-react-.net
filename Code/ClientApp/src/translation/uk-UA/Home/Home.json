{
  "header": "Головна",
  "addNewChart": "Додати графік"
}