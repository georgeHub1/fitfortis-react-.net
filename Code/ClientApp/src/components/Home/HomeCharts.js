import { connect } from 'react-redux';
import { injectIntl } from 'react-intl';
import { getChartIds } from '../../redux/metricsChart.reducer';
import PropTypes from 'prop-types';
import React, { Component } from 'react';
import { bindActionCreators } from 'redux';
import { loadChartBulkAsync } from '../../redux/metricsChart.action';
import { loadMetricsAsync, loadMetricDataAsync, addBackEntry } from '../../redux/metricsMetric.action';
import auth from '../../backendServices/auth';
import { createHomeNewEntryAsync } from '../../redux/metricsMetric.action';
import HomeUnitBodyWeitght from './HomeUnitBodyWeight';
import HomeUnitStressLevel from './HomeUnitStressLevel';
import HomeUnitPulse from './HomeUnitPulse';
import HomeUnitCovid from './HomeUnitCovid-19';

export class HomeCharts extends Component {
  static propTypes = {
    selectedCharts: PropTypes.array.isRequired,
    loadMetrics: PropTypes.func.isRequired,
    loadChartBulk: PropTypes.func.isRequired,
    loadMetricDataAsync: PropTypes.func.isRequired,
    intl: PropTypes.shape({ formatMessage: PropTypes.func.isRequired })
  };
  displayName = HomeCharts.name;
  componentDidMount () {
    const { account } = this.props;

    if (auth.isAuthenticated() && account && account.id) {
      this.props.loadMetrics().then(() => {
        this.props.loadChartBulk();
      });
    }
  }

  componentWillReceiveProps (nextProps) {
    if (auth.isAuthenticated() && this.props.account.id !== nextProps.account.id) {
      setTimeout(() => {
        this.props.loadMetrics().then(() => {
          this.props.loadChartBulk();
        });
      }, 1000);
    }
  }

  render () {
    return (
      <>
        <div className="MainhomeRight" ref={'mainHome'}>
          {auth.isAuthenticated()
          && <div className="homeRight">
              <div className="weight_units">
                <HomeUnitCovid />
              </div>
              <div className="weight_units">
                <HomeUnitBodyWeitght />
              </div>
              <div className="weight_units">
                <HomeUnitStressLevel />
              </div>
              <div className="weight_units">
                <HomeUnitPulse />
              </div>
            </div>
          }
        </div>
      </>
    );
  }
}

const mapStateToProps = state => {
  return {
    selectedCharts: getChartIds(state),
    account: state.profile.account,
    Metric: state.metricsMetric,
    MetricData: state.metricsMetricData
  };
};

const mapDispatchToProps = dispatch => ({
  createNewEntry: bindActionCreators(createHomeNewEntryAsync, dispatch),
  loadChartBulk: bindActionCreators(loadChartBulkAsync, dispatch),
  loadMetrics: bindActionCreators(loadMetricsAsync, dispatch),
  addBackEntry: bindActionCreators(addBackEntry, dispatch),
  loadMetricDataAsync: bindActionCreators(loadMetricDataAsync, dispatch)
});

export default connect(mapStateToProps, mapDispatchToProps)(injectIntl(HomeCharts));
