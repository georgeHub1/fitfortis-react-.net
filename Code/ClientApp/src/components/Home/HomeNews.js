import React, { Component } from 'react';
import { Input, Button, Icon, Typography, Menu, Dropdown, notification } from 'antd';
import { injectIntl } from 'react-intl';
import styles from './styles.module.less';
import { connect } from 'react-redux';
import { bindActionCreators } from 'redux';
import { ReactComponent as share } from '../../img/share.svg';
import { ReactComponent as like } from '../../img/news/like_icon.svg';
import ReactLoading from 'react-loading';
import { data, filter } from './const';
import { Link } from 'react-router-dom';
import { setActiveItem } from '../../redux/item.action';
import { fetchNewsFeedListing, likeNewsFeedItem, unLikeNewsFeedItem, saveNewsSearchItem } from '../../redux/adminTool.action';
import { SanitizeHTML } from '../../components/SanitizeHTML/SanitizeHTML';
import { localeSet } from '../../redux/locale.action';
import analyticId from '../../utils/AnalyticsAndAbTests.tsx';
const { Title } = Typography;
const { Paragraph } = Typography;

class HomeNews extends Component {
  constructor (props) {
    super(props);
    this.state = {
      list: data,
      searchText: '',
      account: props.account,
      abTestUser: '',
      focused: false
    };
  }

  componentWillMount () {
    const { account, intl: { locale }, adminTool: { isNewsFeedLoaded } } = this.props;

    if (window.location.search.includes('?lang=')) {
      const lan = window.location.search.includes('?lang=') ? window.location.search.substring(6, 11) : (localStorage.getItem('language') || 'en');

      this.props.onUpdateLanguage(lan);
      localStorage.setItem('language', lan);
    }
    if (!isNewsFeedLoaded) {
      this.props.fetchNewsFeedListing(account.language || locale);
    }
    const intervalId = setInterval(() => {
      const type =  analyticId.abTest_isArticlesInRandomOrder();

      if (type !== undefined) {
        clearInterval(intervalId);
        if (type === true) {
          this.setState({abTestUser: true });
        } else {
          this.setState({ abTestUser: false });
        }
      }
    }, 100);
  }

  componentWillReceiveProps (nextProps) {
    if (this.props.account.id !== nextProps.account.id) {
      const { intl: { locale }, adminTool: { isNewsFeedLoaded } } = this.props;
      const { account } = nextProps;

      if (window.location.search.includes('?lang=')) {
        const lan = window.location.search.includes('?lang=') ? window.location.search.substring(6, 11) : (localStorage.getItem('language') || 'en');

        this.props.onUpdateLanguage(lan);
        localStorage.setItem('language', lan);
      }
      if (!isNewsFeedLoaded) {
        this.props.fetchNewsFeedListing(account.language || locale);
      }
    }
  }

  getHighlightedText = (higlight, text) => {
    const parts = text.split(new RegExp(`(${higlight})`, 'gi'));

    return (
      <span key={text}>
        {parts.map((part, i) => (
          <span
            key={i}
            // CUSTOMIZATION
            style={
              // CUSTOMIZATION: colorization of search items in Documents
              higlight !== '' && part.toLowerCase().indexOf(higlight.toLowerCase()) !== -1
                ? { fontWeight: 'bold', backgroundColor: '#FCABF9' }
                : {}
            }
          >
            {part}
          </span>
        ))}
      </span>
    );
  };

  copyToClipboard = (url = window.location.href) => {
    const textField = document.createElement('textarea');

    textField.innerText = url;
    document.body.appendChild(textField);
    textField.select();
    document.execCommand('copy');
    textField.remove();
  };
  shareFacebook = (url = window.location.href) => {
    analyticId.firebaseAnalyticsLog('Home_NewsfeedEntry_ClickShareOnHomePage', {shareTo: 'shareToFacebook'});
    return window.open(
      `http://facebook.com/sharer.php?s=100&p[url]=${url}`,
      'fbshare',
      'height=380,width=660,resizable=0,toolbar=0,menubar=0,status=0,location=0,scrollbars=0'
    );
  };
  shareEmail = (url = window.location.href) => {
    analyticId.firebaseAnalyticsLog('Home_NewsfeedEntry_ClickShareOnHomePage', {shareTo: 'shareViaEmail'});
    return window.open(
      `${'mailto:?cc=&subject=Share a link&body='}${url}`,
      '_self'
    );
  };

  likeItem = id => {
    const { account } = this.props;

    this.props.likeNewsFeedItem(account.id, id);
  }

  unLikeItem = id => {
    const { account } = this.props;

    this.props.unLikeNewsFeedItem(account.id, id);
  }

  onBlur = () => {
    this.setState({ focused: false }, () => { this.onChange(); });
  }

  onFocus = () => {
    this.setState({ focused: true });
  }

  onChange = () => {
    const { searchText, focused } = this.state;
    const { adminTool: { newsFeeds } } = this.props;
    const filteredData = filter(newsFeeds, searchText);

    if (filteredData.length === 0 && searchText !== '' && focused) {
      this.addNewSearch(searchText, '', false);
    } else if (!focused && searchText !== '' && filteredData.length !== 0) {
      this.addNewSearch(searchText, filteredData[0].id, true);
    }
  }

  addNewSearch = (title, uniqueId, isSuccessful) => {
    const { id } = this.props.account;

    this.props.saveNewsSearchItem(title, uniqueId, id, isSuccessful);
  };
  shuffle = list =>  {
    return list.reduce((p, n) => {
      const size = p.length;
      const index = Math.trunc(Math.random() * (size - 1));

      p.splice(index, 0, n);
      return p;
    }, []);
  };

  render () {
    const {
      intl: { formatMessage, locale },
      adminTool: { isFetching, newsFeeds }
    } = this.props;
    const { searchText, abTestUser } = this.state;

    let filteredData = filter(newsFeeds, searchText);


    if (abTestUser === true) {
      filteredData = this.shuffle(filteredData);
    }
    const openNotification = () => {
      notification.success({
        message: formatMessage({ id: 'HomeNews.copied' })
      });
    };
    const menu = shareUrl => (
      <Menu className="dropDownShareMenu">
        <Menu.Item onClick={() => this.shareFacebook(shareUrl)}>
          <div className={styles.menuItemBlock}>
            <Icon type="facebook" style={{ marginRight: 10 }} />
            <span>{formatMessage({ id: 'HomeNews.shareToFacebook' })}</span>
          </div>
        </Menu.Item>
        <Menu.Item onClick={() => this.shareEmail(shareUrl)}>
          <div className={styles.menuItemBlock}>
            <Icon type="mail" style={{ marginRight: 10 }} />
            <span>{formatMessage({ id: 'HomeNews.shareToEmail' })}</span>
          </div>
        </Menu.Item>
        <Menu.Item
          onClick={() => {
            analyticId.firebaseAnalyticsLog('Home_NewsfeedEntry_ClickShareOnHomePage', {shareTo: 'copyLink'});
            openNotification();
            this.copyToClipboard(shareUrl);
          }}
        >
          <div className={styles.menuItemBlock}>
            <Icon type="paper-clip" style={{ marginRight: 10 }} />
            <span>{formatMessage({ id: 'HomeNews.copyLink' })}</span>
          </div>
        </Menu.Item>
      </Menu>
    );

    return (
      <div className="home-container">
        <div className="searchBox">
          <Input
            type="text"
            id="searchbox"
            aria-label="search box"
            className={styles.searchInput}
            placeholder={formatMessage({ id: 'HomeNews.search' })}
            onChange={e => {
              this.setState({ ...this.state, searchText: e.target.value }, () => { this.onChange(); });
            }}
            prefix={
              <Icon type="search" className={styles.certainCategoryIcon} />
            }
            onBlur={this.onBlur}
            onFocus={this.onFocus}
          />
        </div>
        <div className={styles.newsList} >
          {
            isFetching ? <ReactLoading className="react-loading-center" type="spin" color="#335889"  />
            : filteredData.map((data, key) => {
              return (
                <div className={styles.newsListItem} key={key}>
                  <Link id={`tabpanel-label-image-text${key}`} onClick={() => this.props.setActiveItem(data)}to={`/newsfeed/${data.id}?lang=${locale}`}>
                    <div className={styles.newsListItemImage}>
                      {
                        data.pictureUrl && <img
                          alt="example"
                          lang={locale}
                          loading="lazy"
                          aria-label="Home news article images"
                          src={data.pictureUrl}
                        />
                      }
                      {
                        data.pictureUrl && <div className={styles.contentDetails}>
                            <div className={styles.contentDetailLeft}>
                              <Title level={2} className={styles.title}><mark>{this.getHighlightedText(searchText, data.title)}</mark></Title>
                            </div>
                          </div>
                      }
                    </div>
                  </Link>
                  <div id="tabpanel-label1" className={styles.newsListItemContent}>
                    <div onClick={() => { this.props.setActiveItem(data); this.props.history.push(`/newsfeed/${data.id}?lang=${locale}`); }}>
                      {
                        !(data.pictureUrl) && <div className={styles.contentDetails}>
                            <div className={styles.contentDetailLeft}>
                              <Title level={2} className={styles.title}>{this.getHighlightedText(searchText, data.title)}</Title>
                            </div>
                          </div>
                      }
                      <Paragraph className={styles.dec}>
                        <SanitizeHTML html={data.description.indexOf('<p>') > -1 ? (data.description.substring(data.description.indexOf('<p>') + 3, data.description.indexOf('</p>')).trim()) : ''} />
                        {formatMessage({ id: 'HomeNews.readMore' })}
                      </Paragraph>
                    </div>
                    <div className={styles.sharePrintBlock}>
                      <Dropdown
                        overlay={menu(`${window.location.href}newsfeed/${data.id}?lang=${locale}`)}
                        trigger={['click']}
                        placement="bottomCenter"
                      >
                        <Button type="primary" aria-label="HomeNewsShare" className={styles.icon}>
                          <Icon component={share} />
                          {formatMessage({ id: 'HomeNews.share' })}
                        </Button>
                      </Dropdown>
                        <Button aria-label="HomeNewsLike" onClick={() => {
                          analyticId.firebaseAnalyticsLog('Home_NewsfeedEntry_ClickLikeOnHomePage', {});
                          if (data.isLiked) {
                            this.unLikeItem(data.id);
                          } else {
                            this.likeItem(data.id);
                          }
                        }} type="primary" className={`${styles.icon}`}>
                          <Icon component={like} className={styles.homeNewslike_btn}/>
                          {`${formatMessage({ id: 'HomeNews.like' })}`}
                        </Button>
                        {data.likes ? <h6>{data.likes} {(data.likes > 1) ? formatMessage({ id: 'HomeNews.manyLikesText' }) : formatMessage({ id: 'HomeNews.oneLikeText' })}</h6> : ''}
                    </div>
                  </div>
                </div>
              );
            })
          }
        </div>
      </div>
    );
  }
}
const mapStateToProps = (({ adminTool, profile: { account } }) => ({ adminTool, account }));

const mapDispatchToProps = dispatch => ({
  setActiveItem: bindActionCreators(setActiveItem, dispatch),
  fetchNewsFeedListing: bindActionCreators(fetchNewsFeedListing, dispatch),
  likeNewsFeedItem: bindActionCreators(likeNewsFeedItem, dispatch),
  unLikeNewsFeedItem: bindActionCreators(unLikeNewsFeedItem, dispatch),
  saveNewsSearchItem: bindActionCreators(saveNewsSearchItem, dispatch),
  onUpdateLanguage (args) {
    dispatch(localeSet(args));
    return Promise.resolve();
  }
});

export default connect(mapStateToProps, mapDispatchToProps)(injectIntl(HomeNews));
