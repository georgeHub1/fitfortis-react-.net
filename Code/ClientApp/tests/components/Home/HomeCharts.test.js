import React from 'react';
import renderer from 'react-test-renderer';
import { MemoryRouter } from 'react-router-dom';
import configureStore from 'redux-mock-store';
import * as chartActions from '../../../src/redux/metricsChart.action';
import { HomeCharts } from '../../../src/components/Home/HomeCharts';
import { Provider } from 'react-redux';
import { initialState } from '../../../src/redux/index';

const mockStore = configureStore([]);

const Container = HomeCharts;

function createNodeMock() {
  const doc = document.implementation.createHTMLDocument();
  return { parentElement: doc.body };
}

describe('HomeCharts', () => {
  it('renders correctly', () => {
    let store = mockStore(initialState);
    store.dispatch(chartActions.loadChartsBulkSuccess({ charts: [], chartMetrics: [], metricData: [], metrics: [] }));

    const wrapper = renderer
      .create(
        <MemoryRouter>
            <Container
              selectedCharts={[]}
              loadMetrics={() => { return Promise.resolve(); }}
              loadChartBulk={() => { return Promise.resolve(); }}
              loadTimeLineData={() => { return Promise.resolve(); }}
              loadMetricDataAsync={() => { return Promise.resolve(); }}
              intl={{ formatMessage: jest.fn() }}
            />
        </MemoryRouter>,
        { createNodeMock }
      )
      .toJSON();

    expect(wrapper).toMatchSnapshot();
  });
});
