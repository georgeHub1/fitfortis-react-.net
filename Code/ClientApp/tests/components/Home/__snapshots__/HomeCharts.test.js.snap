// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HomeCharts renders correctly 1`] = `
<div
  className="MainhomeRight"
/>
`;
